mlprimitives/timeseries.py

```diff
@@ -0,0 +1,63 @@
+import pandas as pd
+import numpy as np
+
+
+def create_window_sequences(df_timeseries, window_size):
+    """
+        Function that takes in a Pandas.DataFrame and a window_size then creates
+            output arrays that correspond to a timeseries sequence with window_size overlap.
+            The output arrays can be fed into a timeseries forecasting model.
+        Inputs:
+            df_timeseries (Pandas.DataFrame): a Pandas dataframe which has 'timestamp'
+                and 'value' columns, and is sorted based on timestamp. 
+                The timestamp column is in UNIX format (in seconds).
+            window_size (int): number of values that overlap to create the sequence.
+        Outputs:
+            x (numpy.ndarray): contains the time series sequenced data.
+            y (numpy.ndarray): acts as the label for the forecasting problem.
+            time (numpy.ndarray): the corresponding timestamps series.
+    """
+    X = []
+    Y = []
+    time = []
+    for i in range(len(df_timeseries) - window_size):
+        X.append(df_timeseries[i: i + window_size]['value'].values.copy().reshape([-1, 1]))
+        Y.append(df_timeseries[i + 1: i + window_size + 1]['value'].values.copy().reshape([-1, 1]))
+        time.append(df_timeseries.iloc[i + window_size]['timestamp'])
+    return np.asarray(X), np.asarray(Y), np.asarray(time)
+
+
+def aggregate_average_time(df_time_value, interval_time_delta, start_time, end_time):
```

Comment:
I would use a more descriptive name like `average_by_time`, or `time_segments_average`.

Reply:
Same thing as above about the `df_` prefix. Also the same for the `_time_delta` suffix. Call them `time_value` and `interval`.
Or, better, rename `time_value` to `X`.

Reply:
Also add `value_column` and `time_column` as input arguments.

Reply:
Also, is it necessary to make start_time and end_time mandatory? What if we just want to process the whole input dataframe, from start to end, without having to worry about which dates we have?
Please make them optional (default=None) and adapt the code accordingly.